src/metabase/driver/ddl/concurrent.clj

```diff
@@ -0,0 +1,18 @@
+(ns metabase.driver.ddl.concurrent
+  (:import [java.util.concurrent Callable Executors ExecutorService Future ThreadFactory]))
+
+(defonce ^:private thread-factory
+  (reify ThreadFactory
+    (newThread [_ r]
+      (doto (Thread. r)
+        (.setName "ddl worker")
+        (.setDaemon true)))))
+
+(defonce ^:private executor
+  (delay (Executors/newFixedThreadPool 3 ^ThreadFactory thread-factory)))
+
+(defn submit-task
```

Comment:
Should this be `submit-task!`? This isn't safe inside a STM transaction -- calling it twice or more is definitely not the same as calling it once

Reply:
Yes that's fair. I honestly don't consider STM at all. I've never used it besides it being the underpinnings of `loaded-libs` and other Clojure internal bits. I treat `!` suffixes as just indicating side effects. I'm honestly not sure how to make submitting tasks to a threadpool safe for STM.

Reply:
The STM thing is really just a thought experiment for cases where it's not clear if a function should have ! or not. Would calling it 10 times instead of just 1 be ok? That's usually the answer to whether it deserves a ! or not.

This is never going to be ok on an STM transaction so it probably deserves a !